### ISSUE TITLE
[BUG] Multiple CVEs in frontend libraries

### ISSUE THREAD

**Issue by DirkSteinsBwi** [jobrunr/jobrunr#112]
**Describe the bug**

The [OWASP dependency check](https://owasp.org/www-project-dependency-check/) reports multiple vulnerabilities in the libraries used by the JobRunr web interface. 

Here's the output of the dependency check goal in maven:

``` 
[ERROR] Failed to execute goal org.owasp:dependency-check-maven:6.1.1:check (default) on project certissuance:
[ERROR]
[ERROR] One or more dependencies were identified with vulnerabilities that have a CVSS score greater than or equal to '5,0':
[ERROR]
[ERROR] package-lock.json?apexcharts: 1625, CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')
[ERROR] package.json: CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')
[ERROR] package-lock.json?elliptic: 1648
[ERROR] package-lock.json: CVE-2020-15133
[ERROR] package-lock.json: CVE-2020-26237
[ERROR] package-lock.json?immer: 1603, CWE-471: Modification of Assumed-Immutable Data (MAID)
[ERROR] package-lock.json?ini: CWE-471: Modification of Assumed-Immutable Data (MAID)
[ERROR] package-lock.json: CVE-2021-28092
[ERROR] package-lock.json: CVE-2020-8203, CVE-2020-28500, CVE-2021-23337
[ERROR] package-lock.json: CVE-2020-28500, CVE-2021-23337
[ERROR] package-lock.json: CVE-2020-7789
[ERROR] package-lock.json?object-path: CWE-471: Modification of Assumed-Immutable Data (MAID), 1573
[ERROR] package-lock.json: CVE-2021-23368
[ERROR] package-lock.json: CVE-2021-23368
[ERROR] package-lock.json?ssri: CVE-2021-27290, 565
[ERROR] package-lock.json: CVE-2021-27290
[ERROR] package-lock.json: CVE-2021-27515
[ERROR] package-lock.json?y18n: 1654
[ERROR]
[ERROR] See the dependency-check report for more details.
```

We use the OWASP dependency check on our CI and the code has to pass to be allowed into production. 

I've attached the generated report for better documentation.  [dependency-check-report.zip](https://github.com/jobrunr/jobrunr/files/6420350/dependency-check-report.zip)

**Environment**
I'm using version version 2.0.0
I'm running on JDK 11.0.8
I'm using the following StorageProvider H2
OWASP dependency check version used is 6.1.1, but the current version6.1.6 doesn't cahnge the report.

**To Reproduce**
Create a simple project and add the OWASP dependency check to the maven or gradle build file. 

**Expected behavior**
No CVEs are in the codebase.

**Additional context**

Even if the frontend can be disabled (which we will do), there shouldn't be known vulnerabilities in the codebase, even if in this case these are in dependent libraries (NodeJS libs). 

Maybe it would be a good idea to separate the UI completely from the JobRunr and have it as a second installable jar which you can add and configure if you want to use it?

**Comment by rdehuyss** [comment]
This will be fixed in release 2.0.1.

Regarding your question to separate the UI/dashboard completely: JobRunr was designed to be as developer-friendly as possible and keeping everything in 1 jar helps with this goal.

When you disable the dashboard, there is no possible way the exploits can be used. I also don't recommend to have the dashboard available publicly but only inside a trusted internal network.

I regularly update all java dependencies of JobRunr to make sure that it stays in great shape - From now on, I will also try to do this more often for the UI/Dashboard and I added an OWASP task to the npm `package.json` build file to avoid such CVE's in the future.

**Comment by DirkSteinsBwi** [comment]
Ok, thanks for your feedback. Looking forward to release 2.0.1
